### DISCUSSION TITLE
hideIf() and buttons

### DISCUSSION THREAD

**Discussion by meavric** [rappasoft/laravel-livewire-tables#1909] [General]
Hi,
I want to show some button if some status = 201.
I Try use:
`
                LinkColumn::make('Kwalifikacja') 
                    ->title(fn($row) => 'Kwalifikacja ')
                    ->location(fn($row) => '#')
                    ->attributes(function($row) {
                        return [
                            'class' => 'btn btn-primary btn-sm',
                            'wire:click.prevent' => 'viewDetailModal('.$row->id.')'
                        ];
                    })
                ->hideIf(fn($row) => ($row->status === 201 ? 1 : 0)),
`
but I have error: Cannot assign Closure to property Rappasoft\LaravelLivewireTables\Views\Column::$hidden of type bool

How can I do this?

BR

**Comment by lrljoe** [comment]
A Column does not accept hideIf based on the row value, as otherwise you'd have a broken DOM.

Otherwise you're saying
"Display a td IF the row status is 201"
So you'd have some rows with 6 TDs, some with 3 TDs, and the entire table would look completely broken.

Use the classes to determine whether or not to display the button, or use a custom view for the Column.

**Comment by meavric** [comment]
How can I make custom view for this column?
In 1 column i want have few button, but no all show in all rows.  I need use if something then show button or no.
Br

**Comment by lrljoe** [comment]
[#](https://rappasoft.com/docs/laravel-livewire-tables/v3/columns/available-methods#content-using-a-view)Using a view
If you would like to render a view for the cell:

```php
Column::make('Name')
    ->format(
        fn($value, $row, Column $column) => view('my.custom.view')->withValue($value)
    ),
```
As a shorthand you can use the following:

```php
Column::make('Name')
    ->view('my.custom.view'),
```
You will have access to $row, $value, and $column from within your view.

You can then have complete control over whether each  of the Buttons is displayed, how they're displayed etc

**Comment by meavric** [comment]
Hi, 
Nice, this is exactly that what I need, but tell me can I get access to other variable than $value, $row, $column ?
BR

**Reply by lrljoe** [reply]
You can use 
```
$this
```
To access the Table Component. But what else do you need for a Column?!

**Comment by meavric** [comment]
Hi,
Yes of course, you have right, $this make all job :)
BR
